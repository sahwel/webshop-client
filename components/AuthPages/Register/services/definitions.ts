export enum Section {
  firstSection = 1,
  secondSection,
  thirdSection,
}
