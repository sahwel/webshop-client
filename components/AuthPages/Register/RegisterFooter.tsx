import Link from 'next/link';
import { FunctionComponent, useCallback, MouseEvent } from 'react';
import Button from '../../common/Form/Button/Button';
import { Section } from './services/definitions';

interface RegisterFooterProps {
  handleSectionChange: (toFoward: boolean) => void;
  section: Section;
}

const RegisterFooter: FunctionComponent<RegisterFooterProps> = ({
  handleSectionChange,
  section,
}) => {
  const sectionChangeToFoward = useCallback(
    (event: MouseEvent<HTMLElement>) => {
      section < Section.thirdSection && event.preventDefault();
      handleSectionChange(true);
    },
    []
  );

  const sectionChangeToBackward = useCallback(() => {
    handleSectionChange(false);
  }, []);

  return (
    <div>
      <Button
        children="Next"
        type={section >= Section.secondSection ? 'submit' : 'button'}
        onClick={sectionChangeToFoward}
      />
      {section === Section.firstSection && (
        <Link href="/login" passHref>
          <a className="text-center block text-sm mt-2 cursor-pointer">Back</a>
        </Link>
      )}
      {section >= Section.secondSection && (
        <p
          onClick={sectionChangeToBackward}
          className="text-center text-sm mt-2 cursor-pointer"
        >
          Back
        </p>
      )}
    </div>
  );
};

export default RegisterFooter;
