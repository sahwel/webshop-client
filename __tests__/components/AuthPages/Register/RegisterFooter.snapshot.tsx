import { render, renderHook, RenderHookResult } from '@testing-library/react';
import RegisterFooter from '@/components/AuthPages/Register/RegisterFooter';
import { Section } from '@/components/AuthPages/Register/services/definitions';

describe('RegisterFooter Snapshot with first section', () => {
  it('renders RegisterFooter with first sectionunchanged', () => {
    const { container } = render(
      <RegisterFooter
        handleSectionChange={function (toFoward: boolean): void {
          throw new Error('Function not implemented.');
        }}
        section={Section.firstSection}
      />
    );
    expect(container).toMatchSnapshot();
  });
});

describe('RegisterFooter Snapshot with second section', () => {
  it('renders RegisterFooter with second sectionunchanged', () => {
    const { container } = render(
      <RegisterFooter
        handleSectionChange={function (toFoward: boolean): void {
          throw new Error('Function not implemented.');
        }}
        section={Section.secondSection}
      />
    );
    expect(container).toMatchSnapshot();
  });
});

describe('RegisterFooter Snapshot with third section', () => {
  it('renders RegisterFooter with third sectionunchanged', () => {
    const { container } = render(
      <RegisterFooter
        handleSectionChange={function (toFoward: boolean): void {
          throw new Error('Function not implemented.');
        }}
        section={Section.thirdSection}
      />
    );
    expect(container).toMatchSnapshot();
  });
});
