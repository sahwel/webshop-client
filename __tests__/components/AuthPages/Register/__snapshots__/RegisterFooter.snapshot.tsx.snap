// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterFooter Snapshot with first section renders RegisterFooter with first sectionunchanged 1`] = `
<div>
  <div>
    <button
      class="button"
      type="button"
    >
      Next
    </button>
    <a
      class="text-center block text-sm mt-2 cursor-pointer"
      href="/login"
    >
      Back
    </a>
  </div>
</div>
`;

exports[`RegisterFooter Snapshot with second section renders RegisterFooter with second sectionunchanged 1`] = `
<div>
  <div>
    <button
      class="button"
      type="button"
    >
      Next
    </button>
    <p
      class="text-center text-sm mt-2 cursor-pointer"
    >
      Back
    </p>
  </div>
</div>
`;

exports[`RegisterFooter Snapshot with third section renders RegisterFooter with third sectionunchanged 1`] = `
<div>
  <div>
    <button
      class="button"
      type="submit"
    >
      Sign up
    </button>
    <p
      class="text-center text-sm mt-2 cursor-pointer"
    >
      Back
    </p>
  </div>
</div>
`;
